includes/Core/McpClient.php

```diff
@@ -0,0 +1,541 @@
+ * Mirrors the McpServer architecture for consistency.
+ */
+class McpClient {
+
+	/**
+	 * Client identifier.
+	 *
+	 * @var string
+	 */
+	private string $client_id;
+
+	/**
+	 * Server URL.
+	 *
+	 * @var string
+	 */
+	private string $server_url;
+
+	/**
+	 * Client configuration.
+	 *
+	 * @var array
+	 */
+	private array $config;
+
+	/**
+	 * Error handler.
+	 *
+	 * @var McpErrorHandlerInterface
+	 */
+	public McpErrorHandlerInterface $error_handler;
+
+	/**
+	 * Observability handler.
+	 *
+	 * @var McpObservabilityHandlerInterface
+	 */
+	public McpObservabilityHandlerInterface $observability_handler;
+
+	/**
+	 * Connection status.
+	 *
+	 * @var bool
+	 */
+	private bool $connected = false;
+
+	/**
+	 * Discovered server capabilities.
+	 *
+	 * @var array
+	 */
+	private array $capabilities = array();
+
+	/**
+	 * MCP session ID for DeepWiki and similar servers.
+	 *
+	 * @var string|null
+	 */
+	private ?string $session_id = null;
+
+	/**
+	 * Transport protocol (sse or mcp).
+	 *
+	 * @var string
+	 */
+	private string $transport = 'mcp';
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string                           $client_id             Client identifier.
+	 * @param string                           $server_url            MCP server URL.
+	 * @param array                            $config                Client configuration.
+	 * @param McpErrorHandlerInterface         $error_handler         Error handler.
+	 * @param McpObservabilityHandlerInterface $observability_handler Observability handler.
+	 */
+	public function __construct(
+		string $client_id,
+		string $server_url,
+		array $config,
+		McpErrorHandlerInterface $error_handler,
+		McpObservabilityHandlerInterface $observability_handler
+	) {
+		$this->client_id             = $client_id;
+		$this->server_url            = rtrim( $server_url, '/' );
+		$this->config                = $config;
+		$this->error_handler         = $error_handler;
+		$this->observability_handler = $observability_handler;
+
+		// Detect transport protocol from URL
+		if ( strpos( $server_url, '/sse' ) !== false ) {
+			$this->transport = 'sse';
+		} elseif ( strpos( $server_url, '/mcp' ) !== false ) {
+			$this->transport = 'mcp';
+		}
+
+
+		// Auto-connect on construction
+		$this->connect();
+	}
+
+	/**
+	 * Connect to the MCP server.
+	 *
+	 * @return bool True on success, false on failure.
+	 */
+	public function connect(): bool {
+		$start_time = microtime( true );
+
+
+		try {
+			// Simple handshake request
+			$response = $this->send_request( 'initialize', array(
+				'protocolVersion' => '2025-06-18',
+				'capabilities'    => array(
+					'roots'    => array( 'listChanged' => true ),
+					'sampling' => array(),
+				),
+				'clientInfo' => array(
+					'name'    => 'WordPress MCP Client',
+					'version' => '1.0.0',
+				),
+			) );
+
+			if ( is_wp_error( $response ) ) {
+				$error_msg = 'MCP Initialize Request Failed: ' . $response->get_error_message();
+				error_log( $error_msg );
```

Comment:
I'm still only partially through my initial code review , but I believe these are handled by the `ErrorLogMcpErrorHandler` abstraction when `$this->error_handler->log() gets called

 - https://github.com/WordPress/mcp-adapter/blob/f41bf0e2c0ea94564a8d45a9bb412c1e33780bf4/includes/Infrastructure/ErrorHandling/ErrorLogMcpErrorHandler.php#L30-L40